test/sequential/test-heapdump-flag-custom-dir.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+const common = require('../common');
+
+
+if (common.isWindows)
+  common.skip('test not supported on Windows');
+
+const assert = require('assert');
+
+const validateDateStringInHeapSnapshotFile = (fileName) => {
+  const currentDate = new Date();
+  const currentYear = currentDate.getFullYear();
+  const currentMonth = (currentDate.getMonth() + 1).toString().padStart(2, '0');
+  const currentDateValue = currentDate.getDate().toString().padStart(2, '0');
+
+  const expectedDateString = `${currentYear}${currentMonth}${currentDateValue}`;
+  const fileDateString = fileName.split('.')[1];
+  assert.deepStrictEqual(fileDateString, expectedDateString);
+};
+
+const validateHeapSnapshotFile = () => {
+  const fs = require('fs');
+
+  assert.strictEqual(process.listenerCount('SIGUSR2'), 1);
+  process.kill(process.pid, 'SIGUSR2');
+  process.kill(process.pid, 'SIGUSR2');
+
+  // Asynchronously wait for the snapshot. Use an async loop to be a bit more
+  // robust in case platform or machine differences throw off the timing.
+  (function validate() {
+    const files = fs.readdirSync(process.cwd());
+
+    if (files.length === 0)
+      return setImmediate(validate);
+
+    assert.strictEqual(files.length, 2);
+
+    for (let i = 0; i < files.length; i++) {
+      assert.match(files[i], /^Heap\..+\.heapsnapshot$/);
+
+      // Check the heapsnapshot file contains the date as YYYYMMDD format
+      validateDateStringInHeapSnapshotFile(files[i]);
```

Comment:
I am not sure if we actually need this, theoretically this could fail if the heap snapshot happens to be generated right before 12:00 AM.

Reply:
Removed.